File: android/app/src/main/java/com/desiora/ai/ui/viewmodel/DesignJobViewModel.kt
package com.desiora.ai.ui.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.desiora.ai.data.model.DesignJobResponse
import com.desiora.ai.data.model.DesignJobListResponse
import com.desiora.ai.data.repository.DesignJobRepository
import com.desiora.ai.data.storage.TokenStorage
import kotlinx.coroutines.launch

class DesignJobViewModel(private val tokenStorage: TokenStorage) : ViewModel() {
    
    private val designJobRepository = DesignJobRepository(tokenStorage)
    
    private val _jobs = MutableLiveData<List<DesignJobResponse>>()
    val jobs: LiveData<List<DesignJobResponse>> = _jobs
    
    private val _currentJob = MutableLiveData<DesignJobResponse?>()
    val currentJob: LiveData<DesignJobResponse?> = _currentJob
    
    private val _isLoading = MutableLiveData<Boolean>()
    val isLoading: LiveData<Boolean> = _isLoading
    
    private val _error = MutableLiveData<String?>()
    val error: LiveData<String?> = _error
    
    fun createJob(
        jobType: String,
        prompt: String,
        imageUrl: String,
        style: String,
        roomType: String?
    ) {
        viewModelScope.launch {
            _isLoading.value = true
            _error.value = null
            
            designJobRepository.createJob(jobType, prompt, imageUrl, style, roomType)
                .onSuccess { job ->
                    _currentJob.value = job
                    _isLoading.value = false
                }
                .onFailure { exception ->
                    _error.value = exception.message
                    _isLoading.value = false
                }
        }
    }
    
    fun getJob(jobId: Int) {
        viewModelScope.launch {
            _isLoading.value = true
            _error.value = null
            
            designJobRepository.getJob(jobId)
                .onSuccess { job ->
                    _currentJob.value = job
                    _isLoading.value = false
                }
                .onFailure { exception ->
                    _error.value = exception.message
                    _isLoading.value = false
                }
        }
    }
    
    fun pollJob(jobId: Int) {
        viewModelScope.launch {
            designJobRepository.getJob(jobId)
                .onSuccess { job ->
                    _currentJob.value = job
                }
        }
    }
    
    fun loadJobs(page: Int = 1, pageSize: Int = 20, status: String? = null) {
        viewModelScope.launch {
            _isLoading.value = true
            _error.value = null
            
            designJobRepository.listJobs(page, pageSize, status)
                .onSuccess { response ->
                    _jobs.value = response.jobs
                    _isLoading.value = false
                }
                .onFailure { exception ->
                    _error.value = exception.message
                    _isLoading.value = false
                }
        }
    }
    
    fun clearError() {
        _error.value = null
    }
}

File: ios/DesioraAI/ViewModels/DesignJobViewModel.swift
import Foundation
import SwiftUI

@MainActor
class DesignJobViewModel: ObservableObject {
    @Published var jobs: [DesignJobResponse] = []
    @Published var currentJob: DesignJobResponse?
    @Published var isLoading = false
    @Published var errorMessage: String?
    
    private let designJobService = DesignJobService.shared
    private let imageService = ImageService.shared
    
    func createJob(
        image: UIImage,
        style: String,
        roomType: String,
        prompt: String
    ) async {
        isLoading = true
        errorMessage = nil
        
        do {
            // Upload image first
            let filename = "\(UUID().uuidString).jpg"
            let imageUrl = try await imageService.uploadImage(image: image, filename: filename)
            
            // Create job
            let job = try await designJobService.createJob(
                jobType: "room_design",
                prompt: prompt.isEmpty ? "Redesign this \(roomType) in \(style) style" : prompt,
                imageUrl: imageUrl,
                style: style,
                roomType: roomType
            )
            
            self.currentJob = job
        } catch {
            self.errorMessage = error.localizedDescription
        }
        
        isLoading = false
    }
    
    func loadJobs() async {
        isLoading = true
        errorMessage = nil
        
        do {
            let response = try await designJobService.listJobs()
            self.jobs = response.jobs
        } catch {
            self.errorMessage = error.localizedDescription
        }
        
        isLoading = false
    }
    
    func getJob(id: Int) async {
        isLoading = true
        errorMessage = nil
        
        do {
            let job = try await designJobService.getJob(id: id)
            self.currentJob = job
        } catch {
            self.errorMessage = error.localizedDescription
        }
        
        isLoading = false
    }
    
    func pollJob(id: Int) async {
        do {
            let job = try await designJobService.getJob(id: id)
            self.currentJob = job
        } catch {
            // Silently fail for polling
        }
    }
}



